src/azure-cli/azure/cli/command_modules/acr/custom.py

```diff
@@ -480,10 +484,10 @@ def remove_identity(cmd, client, registry_name, identities, resource_group_name=
         raise CLIError("The registry {} has no system or user assigned identities.".format(registry_name))
 
     if remove_system_identity:
-        if registry.identity.type == ResourceIdentityType.user_assigned:
+        if registry.identity.type.lower() == ResourceIdentityType.user_assigned.lower():
```

Comment:
The user can't affect the casing in the resource type. So the lower isn't necessary.

Reply:
I think, it is probably still safe to do case insensitive comparison

Reply:
It's found in the test that `registry.identity.type` is always `userAssigned` / `systemAssigned`. But in ResourceIdentityType, the first letter is capitalized.
There may be a problem with the SDK. It takes more time for SDK to eliminate bugs. so it's safe to do case insensitive comparison in cli.